### PR TITLE
[Merged by Bors] - feat: port LinearAlgebra.Finsupp

### PR DESCRIPTION
**Pull request by jcommelin**
---

- [x] depends on: #2336
- [x] depends on: #2347

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Finsupp.lean [hunk 1]
```diff
@@ -1189,25 +1184,33 @@ theorem splittingOfFinsuppSurjective_splits (f : M →ₗ[R] α →₀ R) (s : S
   rw [sum_single_index, one_smul]
   · exact (s (Finsupp.single x 1)).choose_spec
   · rw [zero_smul]
-#align linear_map.splitting_of_finsupp_surjective_splits LinearMap.splittingOfFinsuppSurjective_splits
+#align
+  linear_map.splitting_of_finsupp_surjective_splits
+  LinearMap.splittingOfFinsuppSurjective_splits
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Finsupp.lean:1189]: Please don't do this. The linter has been configured to allow long `#align` statements. And it makes them much more greppable if they are in fact on one line.

**Comment by Komyyy** [comment]
I created the PR #2336 to port this file.

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2336
* leanprover-community/mathlib4#2347
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2336~~
* ~~leanprover-community/mathlib4#2347~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**